sdk/core/core/src/az_span.c

```diff
@@ -176,55 +177,41 @@ AZ_NODISCARD az_result az_span_find(az_span source, az_span target, az_span* out
       {
         // All bytes in `target` have been checked and matched the corresponding bytes in `source` (from the start point `i`),
         // so this is indeed an instance of `target` in that position of `source` (step 4.).
-        // Here we create a span in `source` to map the occurrence of `target` in `source` (pardon the redundancy of words, done for explicit clarity).
-        *out_span = az_span_init(source_ptr + i, target_length, target_length);
 
-        return AZ_OK;
+        return i;
       }
     }
   }
 
   // If the function hasn't returned before, all positions 
   // of `source` have been evaluated but `target` could not be found.
-  return AZ_ERROR_ITEM_NOT_FOUND;
+  return AZ_SPAN_FIND_TARGET_NOT_FOUND;
 }
 
 AZ_NODISCARD az_span az_span_token(az_span source, az_span delimiter, az_span* out_remainder)
 {
   AZ_PRECONDITION_VALID_SPAN(delimiter, 1, false);
   AZ_PRECONDITION_NOT_NULL(out_remainder);
 
-  if (az_span_is_content_equal(source, AZ_SPAN_NULL))
+  if (az_span_length(source) == 0)
   {
     return AZ_SPAN_NULL;
   }
   else
   {
-    uint8_t* source_ptr = az_span_ptr(source);
-    int32_t source_length = az_span_length(source);
-    int32_t source_capacity = az_span_capacity(source);
+    int32_t index = az_span_find(source, delimiter);
 
-    az_span instance;
-    if (az_span_find(source, delimiter, &instance) == AZ_OK)
+    if (index != -1)
```

Comment:
if (index != AZ_SPAN_FIND_TARGET_NOT_FOUND)

Reply:
We are getting rid of that define.